subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoReportBase.java

```diff
@@ -48,6 +51,7 @@
 @Incubating
 public abstract class JacocoReportBase extends JacocoBase {
 
+    private Set<JacocoTaskExtension> taskExtensions = new HashSet<JacocoTaskExtension>();
```

Comment:
I just followed the source and saw that `getAllClassDirs()` is what ultimately ends-up being passed to the ant task. I could have changed a bunch of signatures to propagate the excludes, but this was easier, and did the job.
Source dirs are only used to to generate html reports, so there is no point excluding them.

I had to implement verification myself, and initially had to duplicate a lot of JacocoReport code. With the new codebase I only had to add a doFirst to the verification task to apply the excludes, so it's already way easier.

Since we're on the subject, I would like to suggest a few enhancements:
- A choice between failure and warning for coverage verification, maybe even have different thresholds for the two.
- It would be nice to have the verification task proactively generate an html report on failure.  I use my own task as a a finalizeBy to verification, but it seems logical to save the extra step and provide as much information as possible, something like "verification failed, see report at ...".
-  Have an additional method that adds a test task to the report, but uses shouldRunAfter instead of mustRunAfter. We use this in some multi-module builds, when want to delegate coverage of one module (a library) to another one (the application).

Those are all minor, and this release will already allow us to do a lot of cleanup, so thank you.